### PR TITLE
Increase timeout for server restart in integration tests.

### PR DESCRIPTION
**Pull request by KochetovNicolai**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: tests/integration/helpers/cluster.py [hunk 1]
```diff
@@ -1259,9 +1261,22 @@ def wait_for_start(self, deadline=None, timeout=None):
...
+            current_deadline = deadline
+            # It is possible that server starts slowly.
+            # If container is running, and there is some progress in log, check connection_timeout.
+            if connection_deadline and current_deadline < connection_deadline and status == 'running':
+                rows_in_log = self.count_in_log(".*").strip()
+                try:
+                    rows_in_log = int(rows_in_log)
+                    if rows_in_log > prev_rows_in_log:
+                        prev_rows_in_log = rows_in_log
+                        current_deadline = connection_deadline
```
Comment (alesapin) [inline tests/integration/helpers/cluster.py:1273]: Strange, but it looks like this `if` is always `true` on the first iteration (`whatever > 0`) and does nothing after the first iteration (`current_deadline` already equal to `connection_deadline`).

Comment (KochetovNicolai) [reply tests/integration/helpers/cluster.py:1273]: > it looks like this if is always true on the first iteration

It is not true if logs are empty.

> `current_deadline` already equal to `connection_deadline`

But we set `current_deadline = deadline` in line 1264 ...

Comment (KochetovNicolai) [reply tests/integration/helpers/cluster.py:1273]: However, it is not good that we wait for too long

Comment (KochetovNicolai) [reply tests/integration/helpers/cluster.py:1275]: Now this code is unreadable :(

File: tests/integration/helpers/cluster.py [hunk 2]
```diff
@@ -802,13 +801,13 @@ def start(self, destroy_dirs=True):
...
 
-            start_deadline = time.time() + 20.0  # seconds
+            start_timeout = 20.0  # seconds
             for instance in self.instances.values():
                 instance.docker_client = self.docker_client
                 instance.ip_address = self.get_instance_ip(instance.name)
 
                 print("Waiting for ClickHouse start...")
-                instance.wait_for_start(start_deadline)
+                instance.wait_for_start(start_timeout)
```
Comment (KochetovNicolai) [inline tests/integration/helpers/cluster.py:810]: BTW it is not the same (deadline was common, now it is separate). But I suppose it is ok

**Review by alesapin — APPROVED** [review]
Clear to me, but quite complex :) BTW, we have several `restart_` methods with different waiting logic :)

**Comment by alesapin** [comment]
I hope I'll fix them https://github.com/ClickHouse/ClickHouse/pull/24249

File: tests/integration/helpers/cluster.py [hunk 3]
```diff
@@ -443,7 +443,7 @@ def restart_instance_with_ip_change(self, node, new_ip):
         run_and_check(self.base_cmd + ["up", "--force-recreate", "--no-deps", "-d", node.name])
         node.ip_address = self.get_instance_ip(node.name)
         node.client = Client(node.ip_address, command=self.client_bin_path)
-        node.wait_for_start(timeout=20.0, connection_timeout=600.0)  # seconds
+        node.wait_for_start(start_timeout=20.0, connection_timeout=600.0)  # seconds
```
Comment (KochetovNicolai) [inline tests/integration/helpers/cluster.py:446]: thank you)

**Comment by alesapin** [comment]
image

:(

**Comment by KochetovNicolai** [comment]
From `Integration tests (thread)`
```
2021-05-19 02:15:41,631 Totally finished tests 1328/1329
2021-05-19 02:15:41,631 Finally all tests done, going to compress test dir
```

Timeout was at 02:23:14

**Pull request merged**